onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -56,10 +56,13 @@ ONNX_OPERATOR_KERNEL_EX(
 
 }  // namespace cuda
 
-thread_local std::unique_ptr<CUDAExecutionProvider::PerThreadContextMap> CUDAExecutionProvider::per_thread_context_map_;
+thread_local std::shared_ptr<CUDAExecutionProvider::PerThreadContextMap> CUDAExecutionProvider::per_thread_context_cache_;
```

Comment:
should follow coding standards regarding thread_locals:
https://google.github.io/styleguide/cppguide.html#thread_local